src/Build/Collections/CopyOnWritePropertyDictionary.cs

```diff
@@ -299,8 +299,7 @@ public bool Equals(CopyOnWritePropertyDictionary<T> other)
         /// </summary>
         bool IDictionary<string, T>.ContainsKey(string key)
         {
-            T discard;
-            return ((IDictionary<string, T>)this).TryGetValue(key, out discard);
+            return ((IDictionary<string, T>)this).TryGetValue(key, out _);
```

Comment:
@cdmihai although it is kind of pointless because the hint @rainersigwald was totally right and improves this further than just the avoided allocation of discard...
what happens with the discard underscore in terms of IL? does it just completely avoid the allocation?

Reply:
I think it's just cleaner to look at and produces almost-identical IL (except that the local variable isn't named). Example IL from https://dotnetfiddle.net/zOkQjl

```
    .locals init ([0] class DataClass V_0)
...
    IL_0001:  ldc.i4.3
    IL_0002:  ldloca.s   V_0
    IL_0004:  call       void Program::Method(int32,
                                              class DataClass&)
```

So it still has to allocate the local and pass it to the method with the `out` param, but it's easier to tell at the call site that we don't care about the answer. I don't know if the JIT does anything clever; I don't see anything about that in [the docs](https://docs.microsoft.com/en-us/dotnet/csharp/discards#calls-to-methods-with-out-parameters).